[v10] Fix uncaught errors in Desktop's Discover flow (#21343)

ibeckermayer opened the pull request:
Backport https://github.com/gravitational/teleport/pull/21343 to branch/v10

@ryanclark @kimlisa some surgery was required here including on non-Desktop-Access parts, I believe because we only backported part of #20829. I've manually tested the Desktop Access flow, please double check that I haven't broken anything in any of the other flows.

kimlisa commented on web/packages/teleport/src/Discover/Desktop/DiscoverDesktops/DiscoverDesktops.tsx:
```diff
@@ -132,12 +135,12 @@ export function DiscoverDesktops(props: State) {
-        const osVersion = desktop.labels.find(
-          label => label.name === 'teleport.dev/os_version'
-        ).value;
+        const os =
+          desktop.labels.find(label => label.name === 'teleport.dev/os')
+            ?.value || 'unknown os';
+        const osVersion =
+          desktop.labels.find(label => label.name === 'teleport.dev/os_version')
+            ?.value || 'unknown version';
```
i think this may be the only fix that needs to be backported. i already backported the fix for the `original` bug (which I introduced while back) which you already tested: https://github.com/gravitational/teleport/pull/21517

the other fixes on this backport, was a consequence of refactoring the `polling` logic where we took out the `timeout` feature. I did not backport the `remove timeout` to `v10` b/c as stated here https://gravitational.slack.com/archives/C04NT17MZ60/p1675892265589959

ibeckermayer replied:
@kimlisa there was one other aspect to the fix as well, see the new HEAD @ 0b853169e5305d0261b453dcb88880442ab3b512

Commit:
Manual backport of Fix uncaught errors in Desktop's Discover flow (#21343)

Only backporting the lines required to fix the bug because this branch
has diverged significantly from master in the relevant files.

Commit:
Merge branch 'branch/v10' into isaiah/backport-21343-branch/v10

kimlisa reviewed (approved):
LGTM

Commit:
Merge branch 'branch/v10' into isaiah/backport-21343-branch/v10
